src/com/amazon/corretto/crypto/provider/AesGcmSpi.java

```diff
@@ -967,17 +967,20 @@ protected int engineDoFinal(final ByteBuffer input, final ByteBuffer output)
     return output.position() - initialPosition;
   }
 
-  private void checkOutputBuffer(final int inputLength, final byte[] output, final int outputOffset)
+  private void checkOutputBuffer(
+      final int inputLength, final byte[] output, final int outputOffset, boolean doFinal)
       throws ShortBufferException {
     if (inputLength < 0 || outputOffset < 0 || outputOffset > output.length) {
       throw new ArrayIndexOutOfBoundsException();
     }
-    if (output.length - outputOffset < getUpdateOutputSize(inputLength)) {
```

Comment:
nice catch! i'm surprised we didn't have a unit test covering this. should we add a case [here](https://github.com/corretto/amazon-corretto-crypto-provider/blob/c2af9193453750db04c31171a51692b926e82693/tst/com/amazon/corretto/crypto/provider/test/AesTest.java#L704) to prevent regressions?